### ISSUE TITLE
Suddenly syntax highlight in JSDoc examples is missing

### ISSUE THREAD

**Issue by ackvf** [microsoft/vscode#85187]
Issue Type: <b>Bug</b>

I have just noticed that the syntax highlight in my JSDoc examples is missing. I am certain that it has worked last week.

Note that the syntax highlight does work in tooltip.

![image](https://user-images.githubusercontent.com/2313018/69227144-a9762380-0b81-11ea-91a3-b3befbe8c5c0.png)

VS Code version: Code 1.40.1 (8795a9889db74563ddd43eb0a897a2384129a619, 2019-11-13T16:49:35.976Z)
OS version: Windows_NT x64 10.0.17763

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz (12 x 2592)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: disabled_off<br>surface_control: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.79GB (4.60GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (38)</summary>

Extension|Author (truncated)|Version
---|---|---
overtype|ada|0.2.0
vscode-css-formatter|aes|1.0.1
Bookmarks|ale|10.5.0
copy-relative-path|ale|0.0.2
svelte-intellisense|ard|0.7.1
path-intellisense|chr|1.4.2
jsrefactor|cms|2.20.5
bracket-pair-colorizer-2|Coe|0.0.28
vscode-quick-select|dba|0.2.8
gitlens|eam|10.2.0
EditorConfig|Edi|0.14.2
prettier-vscode|esb|3.8.0
vscode-svelte-snippets|fiv|0.3.0
gc-excelviewer|Gra|2.1.32
vscode-settings-cycler|hoo|1.0.1
relativepathrefactor|jak|0.9.0
svelte-vscode|Jam|0.9.3
jbockle-format-files|jbo|3.0.0
auto-dark-mode|Lin|0.1.6
rainbow-csv|mec|1.3.1
dotenv|mik|1.0.1
vscode-open-in-sublime-merge|mor|0.6.0
vscode-docker|ms-|0.8.2
python|ms-|2019.11.49689
remote-wsl|ms-|0.40.3
vscode-typescript-tslint-plugin|ms-|1.2.3
debugger-for-chrome|msj|4.12.1
material-icon-theme|PKi|3.9.1
vscode-graphql|Pri|0.2.2
multi-command|ryu|1.3.0
code-settings-sync|Sha|3.4.3
vscode-csstree|sme|0.8.0
language-stylus|sys|1.11.0
transpose|v4r|0.0.5
stylint|vtf|0.0.2
vscode-import-cost|wix|2.12.0
sort-js-object-keys|zen|1.0.6
vscode-open-in-github|ziy|1.3.6

(2 theme extensions excluded)

</details>

**Comment by egamma** [comment]
@mjbvz I've also assigned @alexandrudima to this issue since he mentioned some ideas on how to make this feature less costly so that it could be enabled out of the box.

**Comment by alexandrudima** [comment]
@mjbvz It is cool that we have injections and that they work. But in this specific case, for using injections:
- we need to load at least 3 files (the TS grammar, the injection grammar and the markdown grammar). 
- I'm not sure if you are aware of it, but there is a large performance hit for using injections. That is because injections cannot target a rule inside a grammar, since rules cannot be named and injections end up targeting produced scopes. But since scopes can be produced dynamically (from the matched regex), it that means that at each and every point we must match the current scope against injections. This is done here -- https://github.com/microsoft/vscode-textmate/blob/92c6d75748b7dcd696536fc15c072a3b923a01b7/src/grammar.ts#L716-L727 . @aeschli who introduced injections to vscode-textmate could detail more if needed.

Would it not make more sense for ` ``` ` blocks inside JS-doc comments to simply add support for them directly inside the TS grammar? The TS grammar already knows how to parse JS-doc comments, and it doesn't appear that ` ``` ` would be so much more difficult...

**Comment by mjbvz** [comment]
This extension now provides JSDoc markdown highlighting: https://marketplace.visualstudio.com/items?itemName=bierner.jsdoc-markdown-highlighting

---

@alexandrudima Thanks for the additional information. I was not aware of the perf impact of injections.

The main reason we removed this injection was that the markdown grammar pulls in almost all of our other language grammars: #77990

I did create an issue against the TS grammar to support highlighting of example code in JSDoc comments (there are very similar to fenced code blocks): https://github.com/microsoft/TypeScript-TmLanguage/issues/693 However it seems sublime doesn't support textMate's `while` rules which are needed to implement this feature

---

For now, I think shipping this functionality in an extension as reasonable workaround. This also users enable/disable the jsdoc highlighting (#65780)
